server/test/document.spec.js

```diff
@@ -0,0 +1,339 @@
+
+describe('Document Api', () => {
+  before((done) => {
+    server.post('/users')
+      .send(testData.regularUser3)
+      .end((err, res) => {
+        token = res.body.token;
+        userId = res.body.userId;
+      });
+    server.post('/users')
+      .send(testData.adminUser3)
+      .end((err, res) => {
+        should(res.body).have.property('token');
+        adminToken = res.body.token;
+      });
+    server.post('/users')
+      .send(testData.regularUser4)
+      .end((err, res) => {
+        should(res.body).have.property('token');
+        user4Token = res.body.token;
+        done();
+      });
+  });
+
+  describe('Create', () => {
+    it('should create new document', (done) => {
+      server.post('/documents')
+      .send(testData.document)
+      .set({ 'x-access-token': token })
+        .end((err, res) => {
+          res.body.should.have.property('document');
+          should(res.body.document.access).equal('public');
+          documentId1 = res.body.document.id;
+          done();
+        });
+    });
+
+    it('should not create document with null field(s)', (done) => {
+      const doc = {
+        title: 'Doc 1',
+      };
+      server.post('/documents')
+      .send(doc)
+      .set({ 'x-access-token': token })
+        .end((err, res) => {
+          should(res.body).have.a.property('error');
+          done();
+        });
+    });
+
+    it('should send an error message for a user not logged in', (done) => {
+      server.post('/documents')
+      .send(testData.document2)
+        .end((err, res) => {
+          res.status.should.equal(401);
+          done();
+        });
+    });
+
+    it('should be able to make document', (done) => {
+      server.post('/documents')
+      .send(testData.document2)
+      .set({ 'x-access-token': token })
+        .end((err, res) => {
+          res.body.should.have.property('document');
+          should(res.body.document.access).equal('private');
+          documentId2 = res.body.document.id;
+          done();
+        });
+    });
+  });
+
+  describe('Find', () => {
+    it('should return document', (done) => {
+      server.get(`/documents/${documentId1}`)
+      .set({ 'x-access-token': token })
+        .end((err, res) => {
+          res.status.should.equal(200);
+          done();
+        });
+    });
+
+    it('should get a private document for an admin', (done) => {
+      server.get(`/documents/${documentId2}`)
+      .set({ 'x-access-token': adminToken })
+        .end((err, res) => {
+          res.status.should.equal(200);
+          done();
+        });
+    });
+
+    it(`should not return private document to a user 
+    who does NOT own the document`, (done) => {
+      server.get(`/documents/${documentId2}`)
+      .set({ 'x-access-token': user4Token })
+        .end((err, res) => {
+          res.status.should.equal(403);
+          done();
+        });
+    });
+
+    it('should return all documents', (done) => {
```

Comment:
I'm guessing here tests if an authorized user gets all documents the user has access to. Hence, this description is inadequate.